### PR TITLE
Adding r-did, r-bmisc, and r-drdid as a package

### PR DESCRIPTION
**Pull request by JNuss71**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

I used the [conda_r_skeleton_helper](https://github.com/bgruening/conda_r_skeleton_helper?tab=readme-ov-file) to expedite the package review process for adding the [did package](https://cran.r-project.org/web/packages/did/).

### CODE REVIEW

File: recipes/r-bmisc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  host:
+    - r-base
+    - r-rcpp
+    - r-rcpparmadillo
+    - r-data.table
+    - r-dplyr
+    - libblas         # [win]
+    - libcblas        # [win]
+    - liblapack       # [win]
+    - liblapacke      # [win]
```
Comment (mfansler) [inline recipes/r-bmisc/meta.yaml:49]: Should only need:

```suggestion
    - libblas         # [win]
    - liblapack       # [win]
```

File: recipes/r-bmisc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  merge_build_host: True  # [win]
+  missing_dso_whitelist:
+    - '*/R.dll'            # [win]
+    - '*/Rblas.dll'        # [win]
+    - '*/Rlapack.dll'      # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
```
Comment (mfansler) [inline recipes/r-bmisc/meta.yaml:25]: I would suggest just to skip windows for now. We are switching to UCRT-based builds and moving to R 4.3/4.4, whereas this is testing the old build tools on R 4.1.

```suggestion
  skip: true  # [win]

```

File: recipes/r-bmisc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    - r-data.table
+    - r-dplyr
+
+test:
+  commands:
+    - $R -e "library('BMisc')"           # [not win]
+    - "\"%R%\" -e \"library('BMisc')\""  # [win]
+
+about:
+  home: https://bcallaway11.github.io/BMisc/, https://github.com/bcallaway11/BMisc
```
Comment (mfansler) [inline recipes/r-bmisc/meta.yaml:65]: Split URLs

```suggestion
  home: https://bcallaway11.github.io/BMisc/
  dev_url: https://github.com/bcallaway11/BMisc
```

File: recipes/r-did/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    - r-pbapply
+    - r-tidyr
+
+test:
+  commands:
+    - $R -e "library('did')"           # [not win]
+    - "\"%R%\" -e \"library('did')\""  # [win]
+
+about:
+  home: https://bcallaway11.github.io/did/, https://github.com/bcallaway11/did/
```
Comment (mfansler) [inline recipes/r-did/meta.yaml:54]: Split URLs:

```suggestion
  home: https://bcallaway11.github.io/did/
  dev_url: https://github.com/bcallaway11/did/
```

File: recipes/r-drdid/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    - r-bmisc >=1.4.1
+    - r-trust
+
+test:
+  commands:
+    - $R -e "library('DRDID')"           # [not win]
+    - "\"%R%\" -e \"library('DRDID')\""  # [win]
+
+about:
+  home: https://psantanna.com/DRDID/, https://github.com/pedrohcgs/DRDID
```
Comment (mfansler) [inline recipes/r-drdid/meta.yaml:42]: Split URLs:

```suggestion
  home: https://psantanna.com/DRDID/
  dev_url: https://github.com/pedrohcgs/DRDID
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for working on this! ❤️ 

I suggest postponing windows build until on the feedstock. Few minor changes and I think this could be merged.

**Comment by JNuss71** [comment]
Thanks for the suggestions! I merged all of them.

**Review by mfansler — APPROVED** [review]
Ready to go! 🚀

**Pull request merged**